src/app/drive/components/UpgradeToDialogs/RestrictedSharingDialog.tsx

```diff
@@ -0,0 +1,31 @@
+import { Dialog } from '@internxt/ui';
+import { useTranslationContext } from 'app/i18n/provider/TranslationProvider';
+
+interface UpgradeToDialog {
+  isDialogOpen: boolean;
+  onCloseDialog: () => void;
+  onAccept: () => void;
+}
+
+export const RestrictedSharingDialog = ({ isDialogOpen, onAccept, onCloseDialog }: UpgradeToDialog) => {
+  const { translate } = useTranslationContext();
+
+  const title = translate('modals.restrictedSharingModal.title');
+  const subtitle = translate('modals.restrictedSharingModal.subtitle');
+  const primaryAction = translate('actions.upgrade');
+  const secondaryAction = translate('actions.cancel');
```

Comment:
This Dialog is identical to the other one, the only thing that changes is the texts. Have you considered creating one called `UpgradeDialog` or similar and passing it the texts and functions that each one requires?

Reply:
Okay, let's do it.